cmd/job.go

```diff
@@ -0,0 +1,388 @@
+			if lr.OrderBy == nil {
+				rb := "requested-at"
+				lr.OrderBy = &rb
+			}
+			selector := sdk.AspectSelector{
+				SchemaPrefix:   JOB_SCHEMA,
+				ListRequest:    *lr,
+				IncludeContent: true,
+			}
+			if jobsJsonFilter != "" {
+				selector.JsonFilter = &jobsJsonFilter
+			}
+			ctxt := context.Background()
+			if list, res, err := sdk.ListAspect(ctxt, selector, CreateAdapter(true), logger); err == nil {
+				switch outputFormat {
+				case "json":
+					return a.ReplyPrinter(res, false)
+				case "yaml":
+					return a.ReplyPrinter(res, true)
+				default:
+					printJobListTable(list, false)
+				}
+				return nil
+			} else {
+				return err
+			}
+		},
+	}
+
+	readJobCmd = &cobra.Command{
+		Use:     "get [flags] job_id",
+		Aliases: []string{"read", "g"},
+		Short:   "Fetch details about a single job",
+		Args:    cobra.ExactArgs(1),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			recordID := GetHistory(args[0])
+			return readDisplayJob(recordID)
+		},
+	}
+
+	createJobCmd = &cobra.Command{
+		Use:   "create [flags] service-id -f job-input|- -a aspect-urn --watch --stream",
+		Short: "Create a new job",
+		Long: `Create a new job by executing the service 'service-id' with the
+input paramters defined in either a provided (json) file or a reference
+to an aspect containing the parameter definitions. If the job definition is
+provided through 'stdin' use '-' as the file name and also include the --format flag`,
+		Args: cobra.ExactArgs(1),
+		RunE: func(cmd *cobra.Command, args []string) (err error) {
+			ctxt := context.Background()
+
+			if fileName == "" && aspectURN == "" {
+				cobra.CheckErr("Missing parameter file '-f job-file|-' or '-a aspectURN'")
+			}
+
+			serviceID := GetHistory(args[0])
+
+			var pyld a.Payload
+			if fileName != "" {
+				if pyld, err = payloadFromFile(fileName, inputFormat); err != nil {
+					cobra.CheckErr(fmt.Sprintf("While reading job file '%s' - %s", fileName, err))
+				}
+			}
+			if aspectURN != "" {
+				j := fmt.Sprintf(CREATE_FROM_ASPECT, aspectURN, serviceID)
+				if pyld, err = a.LoadPayloadFromBytes([]byte(j), false); err != nil {
+					cobra.CheckErr(fmt.Sprintf("While reading job file '%s' - %s", fileName, err))
+				}
+			}
+			res, err := sdk.CreateServiceJobRaw(ctxt, serviceID, pyld, 0, CreateAdapter(true), logger)
+			if err != nil {
+				return err
+			}
+			if res.StatusCode() == 202 {
+				return waitForResult(ctxt, res, serviceID)
+			}
+			reply, err := res.AsObject()
+			if err != nil {
+				return err
+			}
+			jobID, ok := reply["job-id"].(string)
+			if !ok {
+				cobra.CheckErr("Cannot find job ID in response")
+			}
+			return readDisplayJob(jobID) // a.ReplyPrinter(res, outputFormat == "yaml")
+		},
+	}
+)
+
+type JobCreateT struct {
+	JobID      string  `json:"job-id"`
+	ServiceID  string  `json:"service-id,omitempty"`
+	RetryLater float64 `json:"retry-later"`
+}
+
+func waitForResult(ctxt context.Context, res a.Payload, serviceID string) error {
```

Comment:
Feel we are doing too much for the `create job` endpoint.   My suggestions are:
- create job just do creating a job(sync), and return an address to do polling for users
- user get status by the polling endpoint (job status)
- optional, user get another endpoint to receive streaming events, such as with `/sse`

Reply:
I broke it up a bit, but I still believe it is more convenient if a single command is sufficient to create a job and monitor it. Most cli tools have some kind of "watch" option if the "thing" they kick off takes a while to finish or may change over time.